src/main/java/uz/ccrew/flightmanagement/repository/FlightScheduleRepository.java

```diff
@@ -2,11 +2,24 @@
 
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.stereotype.Repository;
 
 import uz.ccrew.flightmanagement.entity.FlightSchedule;
 
 @Repository
 public interface FlightScheduleRepository extends BasicRepository<FlightSchedule, Long> {
     Page<FlightSchedule> findByOriginAirport_AirportCode(String airportCode, Pageable pageable);
+
+    @Query("SELECT DISTINCT fs FROM FlightSchedule fs " +
+            "JOIN Leg l ON fs.flightNumber = l.flightSchedule.flightNumber " +
+            "WHERE fs.departureDateTime = l.actualDepartureTime " +
+            "AND fs.arrivalDateTime = l.actualArrivalTime")
```

Comment:
```suggestion
            "AND fs.arrivalDateTime = l.actualArrivalTime
            and l.originAirport = fs.originAirportCode")
```
pasdagi queryga ham qoshish kere, keyin beautify qivorin, lowercase da yozin queryni